### DISCUSSION TITLE
Any way for different shapes in flowpart to exchange data amongst them ?

### DISCUSSION THREAD

**Discussion by giannik** [OrchardCMS/OrchardCore#9992] [Q&A]
Hey, 
I am building a complex layout in flowpart with nested widgets , a dashboard kind of page.

At the lowest level of the hierarchy  is  a widget which is a list of content items. 
At the higest level are some filter widgets for setting filters for the results widgets.

Becuase the flowpart renders from highest to lowest widget its not possible for the filter widgets to know about the  results of the list items widget. 

Is there any way to have some sort of pub-sub communication between widget shapes . So that the shape of the results widget can notify subscribers (eg the filter widgets) that it has built itsself and the filter widget shapes can grab  some information passed in from the results widget shape ?

**Comment by ns8482e** [comment]
May be you can use `Aspect`. Implement `handler`  for given `Aspect` and call  content manager's  `PopulateAspectAsync` from Driver or Controller.
